common/network-common/src/main/java/org/apache/spark/network/crypto/AuthClientBootstrap.java

```diff
@@ -97,6 +97,8 @@ public void doBootstrap(TransportClient client, Channel channel) {
       }
       doSaslAuth(client, channel);
     }
+
+    client.setClientId(appId);
```

Comment:
This seems to cause `IllegalStateException: Client ID has already been set` in `AuthIntegrationSuite`.
```
  public void setClientId(String id) {
    Preconditions.checkState(clientId == null, "Client ID has already been set.");
    this.clientId = id;
  }
```
Do we need to check with `getClientId`?